### PR TITLE
(#176) HttpClientEnvelope + PlainHttpClient

### PR DESCRIPTION
**Pull request by llorllale**
This PR is for #176:
* Added `HttpClientEnvelope`
* Refactored existing http clients to extend `HttpClientEnvelope`
* Added `PlainHttpClient`
* Added ctor for `RemoteDocker` that defaults to the `PlainHttpClient`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#189](https://github.com/amihaiemil/docker-java-api/issues/189) is now in scope, role is `REV`

File: src/main/java/com/amihaiemil/docker/HttpClientEnvelope.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.conn.ClientConnectionManager;
+import org.apache.http.params.HttpParams;
+import org.apache.http.protocol.HttpContext;
+
+/**
+ * HTTP client envelope.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @since 0.0.4
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/HttpClientEnvelope.java]: @llorllale ``@version $Id$`` tag here

File: src/main/java/com/amihaiemil/docker/PlainHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+package com.amihaiemil.docker;
+
+import org.apache.http.impl.client.HttpClients;
+
+/**
+ * Plain HTTP (no TLS) client.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @since 0.0.4
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/PlainHttpClient.java]: @llorllale ``@version`` as well here

File: src/main/java/com/amihaiemil/docker/HttpClientEnvelope.java (hunk 1 above)
Comment (llorllale) [reply src/main/java/com/amihaiemil/docker/HttpClientEnvelope.java]: @amihaiemil try using [WriteTag](http://checkstyle.sourceforge.net/config_javadoc.html#WriteTag) to check for this

**Comment by llorllale** [comment]
@amihaiemil added `@version`

> do we use SslHttpClient for anything anymore?

`RemoteDocker`

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**